packages/reakit/src/Step/StepContainer.tsx

```diff
@@ -0,0 +1,256 @@
+/* eslint-disable react/no-unused-prop-types */
+import * as React from "react";
+import * as PropTypes from "prop-types";
+import { Container, ContainerProps, SelectorMap } from "constate";
+import omit from "../_utils/omit";
+
+export const initialState = {
+  loop: false,
+  ids: [],
+  current: -1,
+  ordered: {}
+};
+
+export interface StepContainerState {
+  loop: boolean;
+  ids: Array<string>;
+  current: number;
+  ordered: {
+    [key: string]: string;
+  };
+}
+
+// Selectors types
+export type GetCurrentId = () => string;
+export type HasPrevious = () => boolean;
+export type HasNext = () => boolean;
+export type IndexOf = (idOrIndex: string | number) => number;
+export type IsCurrent = (idOrIndex: string | number) => boolean;
+
+// Action types
+export type Show = (
+  idOrIndex: string | number
+) => (state: StepContainerState) => { current: number };
+
+export type Hide = () => (state: StepContainerState) => { current: number };
+
+export type Toggle = (
+  idOrIndex: string | number
+) => (state: StepContainerState) => { current: number };
+
+export type Previous = () => (
+  state: StepContainerState
+) => { current?: number };
+
+export type Next = () => (state: StepContainerState) => { current?: number };
+export type Reorder = (
+  id: string | number,
+  order?: number
+) => (
+  state: StepContainerState
+) => {
+  ordered: object;
+  ids: Array<string>;
+  current?: number;
+};
+
+export type Register = (
+  id: string | number,
+  order?: number
+) => (
+  state: StepContainerState
+) => {
+  ordered: object;
+  ids: Array<string>;
+  current?: number;
+};
+
+export type Unregister = (
+  id: string
+) => (
+  state: StepContainerState
+) => Partial<StepContainerState> & { current?: number };
+
+export type Update = (
+  id: string,
+  nextId: string,
+  orderArg?: number
+) => (
+  state: StepContainerState
+) => Partial<StepContainerState & { current?: number }>;
```

Comment:
Action types are the public API (the methods that will be passed to `Step`). They should only receive arguments and return void.

Example:
https://github.com/reakit/reakit/blob/0dd2b972abf909c6b66773e435ff991bcb0ca9cc/packages/reakit/src/Hidden/HiddenContainer.tsx#L9-L13